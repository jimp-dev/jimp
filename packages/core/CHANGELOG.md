# v0.16.3 (Sat Feb 04 2023)

#### üêõ Bug Fix

- Fix edgeHandling types [#1080](https://github.com/jimp-dev/jimp/pull/1080) ([@domdomegg](https://github.com/domdomegg))
- upgrade file-type [#1108](https://github.com/jimp-dev/jimp/pull/1108) ([@krudos](https://github.com/krudos))

#### ‚ö†Ô∏è Pushed to `master`

- try this ([@hipstersmoothie](https://github.com/hipstersmoothie))
- update linting ([@hipstersmoothie](https://github.com/hipstersmoothie))
- upgrade prettier ([@hipstersmoothie](https://github.com/hipstersmoothie))

#### Authors: 3

- [@krudos](https://github.com/krudos)
- Adam Jones ([@domdomegg](https://github.com/domdomegg))
- Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie))

---

# v0.14.0 (Mon Jun 29 2020)

#### üöÄ Enhancement

- include Addition (Add) blending mode + Officially drop support for Node 8 [#904](https://github.com/oliver-moran/jimp/pull/904) ([@GlitchyPSIX](https://github.com/GlitchyPSIX))

#### Authors: 1

- GlitchyPSI ([@GlitchyPSIX](https://github.com/GlitchyPSIX))

---

# v0.11.0 (Fri May 15 2020)

#### üöÄ Enhancement

- Removed Core-JS as a dependency. [#882](https://github.com/oliver-moran/jimp/pull/882) ([@EricRabil](https://github.com/EricRabil))

#### üêõ Bug Fix

- Make callback optional for Jimp.rgbaToInt [#889](https://github.com/oliver-moran/jimp/pull/889) ([@HanKruiger](https://github.com/HanKruiger))

#### Authors: 2

- Eric Rabil ([@EricRabil](https://github.com/EricRabil))
- Han Kruiger ([@HanKruiger](https://github.com/HanKruiger))

---

# v0.10.2 (Tue Apr 14 2020)

#### üêõ Bug Fix

- Rewrite handling EXIF orientation ‚Äî add tests, make it plugin-independent [#875](https://github.com/oliver-moran/jimp/pull/875) ([@skalee](https://github.com/skalee))

#### Authors: 1

- Sebastian Ska≈Çacki ([@skalee](https://github.com/skalee))

---

# v0.10.0 (Mon Mar 30 2020)

#### üöÄ Enhancement

- Properly split constructor and instance types [#867](https://github.com/oliver-moran/jimp/pull/867) ([@forivall](https://github.com/forivall))

#### Authors: 1

- Emily Marigold Klassen ([@forivall](https://github.com/forivall))

---

# v0.9.6 (Wed Mar 18 2020)

#### üêõ Bug Fix

- Relax mkdirp dependency to allow newer minimist [#857](https://github.com/oliver-moran/jimp/pull/857) ([@Den-dp](https://github.com/Den-dp))

#### üè† Internal

- Fix TypeScript error on 'next' [#858](https://github.com/oliver-moran/jimp/pull/858) ([@crutchcorn](https://github.com/crutchcorn))

#### Authors: 2

- Corbin Crutchley ([@crutchcorn](https://github.com/crutchcorn))
- Denis Bendrikov ([@Den-dp](https://github.com/Den-dp))

---

# v0.9.3 (Tue Nov 26 2019)

#### üêõ Bug Fix

- `@jimp/cli`, `@jimp/core`, `@jimp/custom`, `jimp`, `@jimp/plugin-blit`, `@jimp/plugin-blur`, `@jimp/plugin-circle`, `@jimp/plugin-color`, `@jimp/plugin-contain`, `@jimp/plugin-cover`, `@jimp/plugin-crop`, `@jimp/plugin-displace`, `@jimp/plugin-dither`, `@jimp/plugin-fisheye`, `@jimp/plugin-flip`, `@jimp/plugin-gaussian`, `@jimp/plugin-invert`, `@jimp/plugin-mask`, `@jimp/plugin-normalize`, `@jimp/plugin-print`, `@jimp/plugin-resize`, `@jimp/plugin-rotate`, `@jimp/plugin-scale`, `@jimp/plugin-shadow`, `@jimp/plugin-threshold`, `@jimp/plugins`, `@jimp/test-utils`, `@jimp/bmp`, `@jimp/gif`, `@jimp/jpeg`, `@jimp/png`, `@jimp/tiff`, `@jimp/types`, `@jimp/utils`
  - Fix regeneratorRuntime errors [#815](https://github.com/oliver-moran/jimp/pull/815) ([@crutchcorn](https://github.com/crutchcorn) [@hipstersmoothie](https://github.com/hipstersmoothie))
- `@jimp/core`
  - Follow redirects [#789](https://github.com/oliver-moran/jimp/pull/789) ([@SaWey](https://github.com/SaWey) sander@solora.be)

#### Authors: 2

- Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie))
- Corbin Crutchley ([@crutchcorn](https://github.com/crutchcorn))
