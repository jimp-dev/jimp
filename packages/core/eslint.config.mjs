import shared from "@jimp/config-eslint/base.js";
export default [
  ...shared,
  {
    rules: {
      "@typescript-eslint/no-explicit-any": "off",
    },
  },
];
