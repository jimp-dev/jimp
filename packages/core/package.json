{
  "name": "@jimp/core",
  "version": "0.3.9",
  "description": "Jimp core",
  "main": "dist/index.js",
  "module": "es/index.js",
  "files": [
    "dist",
    "es",
    "jimp.d.ts",
    "fonts"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/oliver-moran/jimp.git"
  },
  "bugs": {
    "url": "https://github.com/oliver-moran/jimp/issues"
  },
  "scripts": {
    "test": "cross-env BABEL_ENV=test mocha --require @babel/register test/**/*.js",
    "test:watch": "npm run test -- --reporter min --watch",
    "test:coverage": "nyc npm run test",
    "build": "npm run build:node:production && npm run build:module",
    "build:watch": "npm run build:node:debug -- -- --watch --verbose",
    "build:debug": "npm run build:node:debug",
    "build:module": "cross-env BABEL_ENV=module babel src -d es --source-maps --config-file ../../babel.config.js",
    "build:node": "babel src -d dist --source-maps --config-file ../../babel.config.js",
    "build:node:debug": "cross-env BABEL_ENV=development npm run build:node",
    "build:node:production": "cross-env BABEL_ENV=production npm run build:node"
  },
  "author": "Oliver Moran <oliver.moran@gmail.com>",
  "license": "MIT",
  "dependencies": {
    "@jimp/utils": "^0.3.9",
    "any-base": "^1.1.0",
    "buffer": "^5.2.0",
    "core-js": "^2.5.7",
    "exif-parser": "^0.1.12",
    "file-type": "^9.0.0",
    "load-bmfont": "^1.3.1",
    "mkdirp": "0.5.1",
    "phin": "^2.9.1",
    "pixelmatch": "^4.0.2",
    "tinycolor2": "^1.4.1"
  },
  "devDependencies": {
    "should": "^13.2.3"
  },
  "xo": false,
  "publishConfig": {
    "access": "public"
  }
}
