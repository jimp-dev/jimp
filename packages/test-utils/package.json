{
  "name": "@jimp/test-utils",
  "version": "0.22.11",
  "description": "Test utils for jimp extensions.",
  "main": "dist/index.js",
  "repository": "jimp-dev/jimp",
  "module": "es/index.js",
  "scripts": {
    "test": "cross-env BABEL_ENV=test mocha --require @babel/register --recursive test  --extension js",
    "test:watch": "npm run test -- --reporter min --watch",
    "test:coverage": "nyc npm run test",
    "build": "npm run build:node:production && npm run build:module",
    "build:watch": "npm run build:node:debug -- -- --watch --verbose",
    "build:debug": "npm run build:node:debug",
    "build:module": "cross-env BABEL_ENV=module babel src -d es --source-maps --config-file ../../babel.config.js",
    "build:node": "babel src -d dist --source-maps --config-file ../../babel.config.js",
    "build:node:debug": "cross-env BABEL_ENV=development npm run build:node",
    "build:node:production": "cross-env BABEL_ENV=production npm run build:node"
  },
  "author": "",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "pngjs": "^6.0.0"
  },
  "devDependencies": {
    "@jimp/custom": "link:../custom"
  }
}
