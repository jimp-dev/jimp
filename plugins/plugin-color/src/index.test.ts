import { expect, test, describe } from "vitest";

import { makeDonutTestImage } from "@jimp/test-utils";
import { createJimp } from "@jimp/core";

import { methods as color } from "./index.js";

const jimp = createJimp({ plugins: [color] });

describe("canvas color transformation", () => {
  const redDonutJGD = makeDonutTestImage(0x00000000, 0xff000088, 0xff0000ff);

  test("can apply more than one color transformation", () => {
    const image = jimp.fromBitmap(redDonutJGD);
    const newJGD = image.color([
      { apply: "hue", params: [-180] },
      { apply: "lighten", params: [25] },
    ]);

    expect(newJGD.bitmap).toMatchSnapshot();
  });

  test("lighten", () => {
    const image = jimp.fromBitmap(redDonutJGD);
    expect(image.color([{ apply: "lighten", params: [25] }])).toMatchSnapshot();
  });

  test("brighten", () => {
    const image = jimp.fromBitmap(redDonutJGD);
    expect(
      image.color([{ apply: "brighten", params: [25] }]),
    ).toMatchSnapshot();
  });

  test("spin hue", () => {
    const image = jimp.fromBitmap(redDonutJGD);
    expect(image.color([{ apply: "hue", params: [150] }])).toMatchSnapshot();
  });
});

describe("brightness", () => {
  test("brightness of 1 should be the same as no change", () => {
    const jgd = makeDonutTestImage(0xff000088, 0xff000088, 0xff000088);
    const image = jimp.fromBitmap(jgd);
    const image2 = jimp.fromBitmap(jgd);

    expect(image.brightness(1).bitmap).toStrictEqual(image2.bitmap);
  });

  test("should be able to brighten", () => {
    const jgd = makeDonutTestImage(0xaa000088, 0xaa000088, 0xaa000088);
    const image = jimp.fromBitmap(jgd);

    expect(image.brightness(2).bitmap).toMatchSnapshot();
  });

  test("should be able to darken", () => {
    const jgd = makeDonutTestImage(0xaa000088, 0xaa000088, 0xaa000088);
    const image = jimp.fromBitmap(jgd);

    expect(image.brightness(0.2).bitmap).toMatchSnapshot();
  });
});
