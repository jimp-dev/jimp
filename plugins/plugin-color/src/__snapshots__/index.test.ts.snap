// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`brightness > should be able to brighten 1`] = `
Visualization:

6666666666
6666666666
6666666666
6666666666
6666666666
6666666666
6666666666
6666666666
6666666666
6666666666

Data:

FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸ FF-00-00⁸⁸
`;

exports[`brightness > should be able to darken 1`] = `
Visualization:

CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC
CCCCCCCCCC

Data:

22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸ 22-00-00⁸⁸
`;

exports[`canvas color transformation > brighten 1`] = `
Visualization:

7777777777
777▫FF▫777
77FFFFFF77
7▫FF▫▫FF▫7
7FF▫77▫FF7
7FF▫77▫FF7
7▫FF▫▫FF▫7
77FFFFFF77
777▫FF▫777
7777777777

Data:

40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ 40-40-40⁰⁰
40-40-40⁰⁰ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ 40-40-40⁰⁰
40-40-40⁰⁰ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ 40-40-40⁰⁰
40-40-40⁰⁰ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40ᶠᶠ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-40-40⁸⁸ FF-40-40ᶠᶠ FF-40-40ᶠᶠ FF-40-40⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
`;

exports[`canvas color transformation > can apply more than one color transformation 1`] = `
Visualization:

7777777777
7779■■9777
77■■■■■■77
79■■99■■97
7■■9779■■7
7■■9779■■7
79■■99■■97
77■■■■■■77
7779■■9777
7777777777

Data:

40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 40-40-40⁰⁰
40-40-40⁰⁰ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 40-40-40⁰⁰
40-40-40⁰⁰ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 40-40-40⁰⁰
40-40-40⁰⁰ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 80-FF-FF⁸⁸ 80-FF-FFᶠᶠ 80-FF-FFᶠᶠ 80-FF-FF⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
`;

exports[`canvas color transformation > lighten 1`] = `
Visualization:

7777777777
777E▵▵E777
77▵▵▵▵▵▵77
7E▵▵EE▵▵E7
7▵▵E77E▵▵7
7▵▵E77E▵▵7
7E▵▵EE▵▵E7
77▵▵▵▵▵▵77
777E▵▵E777
7777777777

Data:

40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ 40-40-40⁰⁰
40-40-40⁰⁰ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ 40-40-40⁰⁰
40-40-40⁰⁰ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ 40-40-40⁰⁰
40-40-40⁰⁰ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80ᶠᶠ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ FF-80-80⁸⁸ FF-80-80ᶠᶠ FF-80-80ᶠᶠ FF-80-80⁸⁸ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰ 40-40-40⁰⁰
`;

exports[`canvas color transformation > spin hue 1`] = `
Visualization:

          
   ▩33▩   
  333333  
 ▩33▩▩33▩ 
 33▩  ▩33 
 33▩  ▩33 
 ▩33▩▩33▩ 
  333333  
   ▩33▩   
          

Data:

00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰
00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰
00-00-00⁰⁰ 00-00-00⁰⁰ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-00-00⁰⁰ 00-00-00⁰⁰
00-00-00⁰⁰ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-00-00⁰⁰
00-00-00⁰⁰ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-00-00⁰⁰
00-00-00⁰⁰ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-00-00⁰⁰
00-00-00⁰⁰ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-00-00⁰⁰
00-00-00⁰⁰ 00-00-00⁰⁰ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-00-00⁰⁰ 00-00-00⁰⁰
00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-FF-80⁸⁸ 00-FF-80ᶠᶠ 00-FF-80ᶠᶠ 00-FF-80⁸⁸ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰
00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰ 00-00-00⁰⁰
`;
